core/src/main/scala/pureconfig/DerivedReaders.scala

```diff
@@ -20,9 +20,26 @@ import shapeless.labelled._
 trait AllowMissingKey { self: ConfigReader[_] => }
 
 /**
+ * Trait extending [[DerivedReaders1]] that contains `ConfigReader` instances for `AnyVal`.
+ *
+ * This trait exists to give priority to the `AnyVal` derivation over the generic product derivation.
+ */
+trait DerivedReaders extends DerivedReaders1 {
```

Comment:
I'm actually not sure what are the implications of using one method instead of the other. I know that a lot of projects working with typeclasses use my approach to disambiguate and I went for it. I don't see an issue except that now we have two traits.